### ISSUE TITLE
Bug: Training Qwen-2.5-omni with GRPO and multimodal (audio+video) input results in 0 loss and empty output

### ISSUE THREAD

**Issue by cheliu-computation** [modelscope/ms-swift#6069]
<p><b>Describe the bug</b>
When fine-tuning the <code>Qwen-2.5-omni</code> model using the GRPO algorithm on a dataset with both audio and visual inputs from the same video source, the training metrics show no signs of learning. The <code>loss</code>, <code>grad_norm</code>, and <code>reward</code> all remain at <code>0.0</code> from the very first step.</p><p>Consequently, when trying to generate output from the model during or after this training process, it produces an empty string. This behavior suggests a fundamental issue in how the data is being processed, how gradients are calculated, or how the reward is being computed for this specific multimodal setup.</p><p><b>To Reproduce</b>
Steps to reproduce the behavior:</p><ol start="1"><li><p>Set up a training environment using the MS-Swift framework with the versions specified below.</p></li><li><p>Configure the training script to use <code>qwen-2.5-omni</code> as the base model.</p></li><li><p>Set the fine-tuning algorithm to <code>sft_type: 'grpo'</code>.</p></li><li><p>Use a custom dataset where each sample contains paired video frames and audio waveforms from a single video.</p></li><li><p>Launch the fine-tuning process.</p></li><li><p>Observe the training logs, which will display the metrics as shown below.</p></li><li><p>Attempt to run inference with the saved checkpoints, which will result in an empty string output.</p></li></ol><p><b>Expected behavior</b>
The expected behavior is that the model should learn from the multimodal inputs. Specifically:</p><ul><li><p>The <code>loss</code> should be a non-zero value and should decrease as training progresses.</p></li><li><p>The <code>reward</code> metric for GRPO should be a meaningful, non-zero value.</p></li><li><p>The <code>grad_norm</code> should be non-zero, indicating that gradients are flowing.</p></li><li><p>The model should be able to generate relevant, non-empty text when prompted.</p></li></ul><p><b>Training Logs</b>
The following log is consistently produced at each step:</p><response-element class="" ng-version="0.0.0-PLACEHOLDER"><!----><!----><!----><!----><code-block _nghost-ng-c3606480851="" class="ng-tns-c3606480851-76 ng-star-inserted" style=""><!----><!----><div _ngcontent-ng-c3606480851="" class="code-block ng-tns-c3606480851-76 ng-animate-disabled ng-trigger ng-trigger-codeBlockRevealAnimation" jslog="223238;track:impression,attention;BardVeMetadataKey:[[&quot;r_971247ca519c39ce&quot;,&quot;c_ed43e9b7d7d0bdac&quot;,null,&quot;rc_5d4af0d990cd3805&quot;,null,null,&quot;en&quot;,null,1,null,null,1,0]]" data-hveid="0" decode-data-ved="1" data-ved="0CAAQhtANahgKEwi62NaD9ZSQAxUAAAAAHQAAAAAQjgI" style="display: block;"><div _ngcontent-ng-c3606480851="" class="code-block-decoration header-formatted gds-title-s ng-tns-c3606480851-76 ng-star-inserted" style=""><span _ngcontent-ng-c3606480851="" class="ng-tns-c3606480851-76">JSON</span><div _ngcontent-ng-c3606480851="" class="buttons ng-tns-c3606480851-76 ng-star-inserted"><button _ngcontent-ng-c3606480851="" aria-label="Copy code" mat-icon-button="" mattooltip="Copy code" class="mdc-icon-button mat-mdc-icon-button mat-mdc-button-base mat-mdc-tooltip-trigger copy-button ng-tns-c3606480851-76 mat-unthemed ng-star-inserted" mat-ripple-loader-uninitialized="" mat-ripple-loader-class-name="mat-mdc-button-ripple" mat-ripple-loader-centered="" jslog="179062;track:generic_click,impression;BardVeMetadataKey:[[&quot;r_971247ca519c39ce&quot;,&quot;c_ed43e9b7d7d0bdac&quot;,null,&quot;rc_5d4af0d990cd3805&quot;,null,null,&quot;en&quot;,null,1,null,null,1,0]];mutable:true"><span class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"></span><mat-icon _ngcontent-ng-c3606480851="" role="img" fonticon="content_copy" class="mat-icon notranslate google-symbols mat-ligature-font mat-icon-no-color" aria-hidden="true" data-mat-icon-type="font" data-mat-icon-name="content_copy"></mat-icon><span class="mat-focus-indicator"></span><span class="mat-mdc-button-touch-target"></span></button><!----><!----></div><!----></div><!----><div _ngcontent-ng-c3606480851="" class="formatted-code-block-internal-container ng-tns-c3606480851-76"><div _ngcontent-ng-c3606480851="" class="animated-opacity ng-tns-c3606480851-76"><pre _ngcontent-ng-c3606480851="" class="ng-tns-c3606480851-76"><code _ngcontent-ng-c3606480851="" role="text" data-test-id="code-content" class="code-container formatted ng-tns-c3606480851-76">{'loss': <span class="hljs-number">0.0</span>, 'grad_norm': <span class="hljs-number">0.0</span>, 'learning_rate': <span class="hljs-number">4e-08</span>, 'reward': <span class="hljs-number">0.0</span>, 'reward_std': <span class="hljs-number">0.0</span>, 'frac_reward_zero_std': <span class="hljs-number">1.0</span>, 'rewards/BoxMatch/mean': <span class="hljs-number">0.0</span>, 'rewards/BoxMatch/std': <span class="hljs-number">0.0</span>, 'completions/mean_length': <span class="hljs-number">1699.9375</span>, 'completions/min_length': <span class="hljs-number">2.0</span>, 'completions/max_length': <span class="hljs-number">4097.0</span>, 'completions/clipped_ratio': <span class="hljs-number">0.375</span>, 'clip_ratio/low_mean': <span class="hljs-number">0.0</span>, 'clip_ratio/low_min': <span class="hljs-number">0.0</span>, 'clip_ratio/high_mean': <span class="hljs-number">0.0</span>, 'clip_ratio/high_max': <span class="hljs-number">0.0</span>, 'clip_ratio/region_mean': <span class="hljs-number">0.0</span>, 'epoch': <span class="hljs-number">0.0</span>, 'global_step/max_steps': '<span class="hljs-number">1</span>/<span class="hljs-number">2219</span>', 'percentage': '<span class="hljs-number">0.05</span>%', 'elapsed_time': '<span class="hljs-number">1</span>m <span class="hljs-number">50</span>s', 'remaining_time': '<span class="hljs-number">2</span>d <span class="hljs-number">19</span>h <span class="hljs-number">56</span>m <span class="hljs-number">55</span>s', 'memory(GiB)': <span class="hljs-number">24.14</span>, 'train_speed(iter/s)': <span class="hljs-number">0.009067</span>}
</code></pre><!----></div></div></div><!----><!----></code-block><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----></response-element><p><b>Environment</b></p><div class="horizontal-scroll-wrapper"><div class="table-block-component"><response-element class="" ng-version="0.0.0-PLACEHOLDER"><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!----><!---->
Component | Version
-- | --
MS-Swift | Current
vLLM | 0.11.0
Python | 3.12
GPU/Hardware | H100 80GB
CUDA Version | 12.8
PyTorch Version | 2.8.0

</div></div></table-block></response-element></div></div>

Additional context

<img width="1632" height="1080" alt="Image" src="https://github.com/user-attachments/assets/a22abbe1-5c2f-4693-b6ce-4d19eda83c70" />

This issue raises a few questions:

Is the combination of Qwen-2.5-omni + GRPO + simultaneous audio + vision inputs officially supported in the current version of MS-Swift? The behavior suggests there might be an incompatibility.

Could this issue also affect the upcoming Qwen3-omni model, or is it specific to Qwen-2.5-omni?

Any insights or potential workarounds would be greatly appreciated.

**Comment by hjh0119** [comment]
It is normal for the loss and grad_norm to be zero if the reward is zero, since the advantages would be zero. Therefore, the immediate cause of this issue is the zero reward.

The zero reward appears to result from an empty completion, which is likely due to issues with length-related parameters. For example, using the default number of video frames or resolution may lead to excessively long visual tokens, which in turn affects `max_new_tokens` (or `max_completion_length`).

Additionally, it could also be caused by a corrupted or incorrect checkpoint, leading to abnormal model outputs.

**Comment by LukeLIN-web** [comment]
Does it fixed? Does qwen3 omni supported GRPO in swift?
